### PR TITLE
Implement SDL-0224 Navigation Subscription Buttons

### LINKED ISSUE #1269: [SDL 0224] Navigation Subscription Buttons
Proposal: [Navigation Subscription Buttons](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0224-navigation-subscription-buttons.md)

> This proposal adds support for new subscription buttons (like the existing Play/Pause, Next, Previous, etc.) specifically designed for navigation scenarios.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/712

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revisions:

- Incorporate [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/712#issuecomment-487981666), with NAV_ROTATE including rotating left or right 90 degrees. 
- Add button press to the Navigation Service's handled RPCs with these new button types.

>Additionally, there was a question in the Steering Committee meeting about requiring applications to be compatible with this proposal.  It was advised that this requirement is out of the scope of SDL, and OEMs would need to ask application developers to implement this proposal once it's available in the library for developers to implement.

> The proposal .md file was updated to reflect these revisions on 5/20/19.

### PR DESCRIPTION
**Pull request by justingluck93**
Fixes #1269 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Summary
Adding enum values so navigation apps can subscribe to navigation hard buttons 

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLButtonName.h [hunk 1]
```diff
@@ -208,3 +208,79 @@ extern SDLButtonName const SDLButtonNameShuffle;
...
+
+/**
+ * Represents a Zoom out button.
+ */
+extern SDLButtonName const SDLButtonNameZoomOut;
+
+/**
+ * Represents a Pan up button
+ */
+extern SDLButtonName const SDLButtonNamePanUP;
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLButtonName.h]: fixed

File: SmartDeviceLink/SDLButtonName.h [hunk 2]
```diff
@@ -208,3 +208,79 @@ extern SDLButtonName const SDLButtonNameShuffle;
...
+#pragma mark - Navigation Buttons
+/**
+ * Represents a Navigate to center button.
+ */
+extern SDLButtonName const SDLButtonNameNavCenter;
+
+/**
+ * Represents a Zoom in button.
+ */
+extern SDLButtonName const SDLButtonNameZoomIn;
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLButtonName.h]: done

File: SmartDeviceLink/SDLButtonName.h [hunk 3]
```diff
@@ -208,3 +208,79 @@ extern SDLButtonName const SDLButtonNameShuffle;
  * Represents a Repeat button.
  */
 extern SDLButtonName const SDLButtonNameRepeat;
+
+#pragma mark - Navigation Buttons
+/**
+ * Represents a Navigate to center button.
+ */
+extern SDLButtonName const SDLButtonNameNavCenter;
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLButtonName.h]: done

File: SmartDeviceLink/SDLButtonName.h [hunk 4]
```diff
@@ -208,3 +208,79 @@ extern SDLButtonName const SDLButtonNameShuffle;
...
+
+/*
+ * Represents a Rotate clockwise button
+ */
+extern SDLButtonName const SDLButtonNameRotateClockwise;
+
+/*
+ * Represents a Rotate counterclockwise button
+ */
+extern SDLButtonName const SDLButtonNameRotateCounterClockWise;
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLButtonName.h]: done

File: SmartDeviceLink/SDLButtonName.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLButtonName.h]: The `SDLButtonNameNavCenter ` name still needs to be updated.

File: SmartDeviceLink/SDLButtonName.h [hunk 5]
```diff
@@ -208,3 +208,79 @@ extern SDLButtonName const SDLButtonNameShuffle;
...
+
+/**
+ * Represents a Pan right button
+ */
+extern SDLButtonName const SDLButtonNameNavPanRight;
+
+/**
+ * Represents a Pan down/right button
+ */
+extern SDLButtonName const SDLButtonNamePanDownRight;
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLButtonName.h]: The button name is missing the `Nav`

**Pull request merged**